### DISCUSSION TITLE
GPUs cannot work simultaneously

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#15670] [Q&A and General discussion]
Hi, I set multi-gpus for milvus-server with command "docker run -d --name milvus_gpu_1.1.1 --gpus device=1,2,3 \
-p 19530:19530 \
-p 19121:19121 \
-v /data0/usr/xiaojuan8/milvus/db:/var/lib/milvus/db \
-v /data0/usr/xiaojuan8/milvus/conf:/var/lib/milvus/conf \
-v /data0/usr/xiaojuan8/milvus/logs:/var/lib/milvus/logs \
-v /data0/usr/xiaojuan8/milvus/wal:/var/lib/milvus/wal \
milvusdb/milvus:1.1.1-gpu-d061621-330cc6"

in server_config.yaml
![image](https://user-images.githubusercontent.com/74544828/154948825-8ebbefcf-9f13-4082-a176-2533cd9ab9e8.png)

When I create IVFPQ index， I found ONLY one gpu is working at the same time. How could all gpus work for indexing at the same time?
![image](https://user-images.githubusercontent.com/74544828/154939166-ca69647e-2dfc-4cff-b415-2f81d5032b69.png)
![image](https://user-images.githubusercontent.com/74544828/154939228-9ddafd90-859f-4dd6-a533-668569048b71.png)

**Comment by xiaojuan8** [comment]
I observe the GPU-util monitor for minutes, gpus are not working simultaneously all the way.

**Reply by yhmo** [reply]
I just check the source code, the build index thread always uses one device to build, no matter CPU mode or GPU mode. It is not a technical limitation, we didn't get a chance to optimize it.

**Reply by xiaojuan8** [reply]
ok,thanks.

**Reply by xiaojuan8** [reply]
hi, today, I rerun my program, I found gpus can work together. That is confusing.
![image](https://user-images.githubusercontent.com/74544828/155056626-d51aa95c-9fc1-4e25-9f61-ec112fa059bf.png)
